An other attempt to fix nosetests on OSX-Conda

bam241 opened the pull request:
@gonuke, I need to confess that I am throwing darts in the dark here.

I am trying to fix a warning that I don't understand, and don't observe on my own machine....
here is the warning:
`/Users/runner/miniforge3/conda-bld/pyne_1597167143946/work/src/utils.h:218:14: warning: address of stack memory associated with local variable 'msgstr' returned [-Wreturn-stack-address]
`
and I used [this](https://flaviocopes.com/c-return-string/) as an inspiration

bam241 committed:
simplify throw return

bam241 committed:
updating changelog

bam241 committed:
other attempt to fix conda test on OSX

bam241 commented:
and btw we are unsure if this is actually the source of the problem....

ngranda commented on src/nucname.h:
```diff
@@ -138,7 +138,8 @@ namespace nucname
         NaNEstr += nucnow;
       }
-      return (const char *) NaNEstr.c_str();
+      const char* NaNEstr_rtn = NaNEstr.c_str();
```
This may be obvious, but why are you keeping `const char* ` in the changes to this file (`nucname.h`) and not in the one for the above file (`h5wrap.h`). The changelog says these may cause some problems.

bam241 replied:
this was not intended.
Let me push a fix...

thx for catching this

bam241 committed:
propagating the change in h5wrap

ngranda commented on src/rxname.h:
```diff
@@ -344,10 +345,11 @@ namespace rxname
     }
 
-  private:
+   private:
```
Is this extra space on purpose?

ngranda reviewed (commented):
I just have one question about the extra blank space, but otherwise LGTM! Thanks @bam241

bam241 committed:
remove space

gonuke commented:
Thanks @bam241 - let's hope this makes a difference

bam241 commented:
@gonuke I PRed this one vs develop :(

gonuke commented:
Oops!  Does that mean we need another PR to test this in conda?

bam241 commented:
No I updated my PR on Pune-feedstock...
I fixed the warning, but not the unexpected failure of the test...
Not sure what to do with OSX-conda....

> On Aug 12, 2020, at 19:58, Paul Wilson <notifications@github.com> wrote:
> 
> ﻿
> Oops! Does that mean we need another PR to test this in conda?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
